src/modules/service-catalog/ServiceCatalogItemPage.tsx

```diff
@@ -97,43 +32,65 @@ export function ServiceCatalogItemPage({
   brandId,
 }: ServiceCatalogItemPageProps) {
   const serviceCatalogItem = useServiceCatalogItem(serviceCatalogItemId);
-  const { requestFields, handleChange } = useItemFormFields(
+  const {
+    requestFields,
+    associatedLookupField,
+    setRequestFields,
+    handleChange,
+  } = useItemFormFields(serviceCatalogItem, baseLocale);
+  const { submitServiceItemRequest } = useServiceFormSubmit(
     serviceCatalogItem,
+    requestFields,
+    associatedLookupField,
     baseLocale
   );
-  const { t } = useTranslation();
+
+  const handleRequestSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const response = await submitServiceItemRequest();
+    if (response && !response.ok) {
+      const errorData = await response.json();
```

Comment:
Nit: it is probably better to create a type for the entire response and cast it here instead of just creating a type for `RequestError` and using it later

Reply:
Ok I added type for response